service/history/configs/config.go

```diff
@@ -358,7 +358,7 @@ func NewConfig(
 
 		EmitShardLagLog:                       dc.GetBoolProperty(dynamicconfig.EmitShardLagLog, false),
 		HistoryCacheInitialSize:               dc.GetIntProperty(dynamicconfig.HistoryCacheInitialSize, 128),
-		HistoryCacheMaxSize:                   dc.GetIntProperty(dynamicconfig.HistoryCacheMaxSize, 512),
+		HistoryCacheMaxSize:                   dc.GetIntProperty(dynamicconfig.HistoryCacheMaxSize, 51200),
```

Comment:
I think this is too small as default. At least 128K which is 250*512. Average mutable state size is 1KB, that will be 128MB which is not bad at all. We should change to use byte size as limit not entity count, but that is for future.

Reply:
Update to 256K